lib/open_api_spex/cast/discriminator.ex

```diff
@@ -93,20 +94,23 @@ defmodule OpenApiSpex.Cast.Discriminator do
     end
   end
 
-  defp find_discriminator_schema(discriminator, _, schemas) do
-    Enum.find(schemas, &Kernel.==(&1.title, discriminator))
-  end
+  defp find_discriminator_schema(discriminator, _mappings, schemas)
+       when is_atom(discriminator) and not is_nil(discriminator),
+       do: Enum.find(schemas, &Kernel.==(&1."x-struct", discriminator))
```

Comment:
Is `Kernel.==` necessary here? can it be:

```suggestion
       do: Enum.find(schemas, & &1."x-struct" ==  discriminator)
```

?